processor/test/integration/projects.test.ts

```diff
@@ -60,20 +61,33 @@ vi.mock('../../src/meili.js', () => {
   meiliIndex.search.mockReturnValue(Promise.resolve(searchResponse))
   return { meiliIndex }
 })
-
-type GiteaDB = typeof giteaDBImported
+type GiteaDB = MockProxy<typeof giteaDBImported> & {
+  addRepoCallback?: (row: Row | null, info: ReplicationEvent) => Promise<void>
+}
```

Comment:
I think we could get rid of this weird type and make it simpler, more readable and more correct by defining a function like this instead (untested): 

```ts
function dispatchRepoEvent(giteaDB, repoInfo, repoEvent) {
  for (const [callback] of giteaDB.subscribeToRepo.mock.calls) {
    callback(repoInfo, repoEvent)
  }
}
```

Reply:
Resolved in 4a19adc.